### ISSUE TITLE
Jaggrab / Http / On demand - Timing out

### ISSUE THREAD

**Issue by thispixel** [apollo-rsps/apollo#138]
On slow connections the server is read timing out the connection before the buffer is written. Causing the client to throw IO exceptions due to unexpected socket closure.

I don't know if removing the IdleStateHandler from the channel pipeline after the request is read will suffice.

**Comment by WizardJesse1** [comment]
@thispixel does anything get said in the console? i think ive seen this happen once

**Comment by thispixel** [comment]
Any large buffer size will cause it to happen, Artificial lag produces the issue consistently.

**Comment by Major-** [comment]
How slow are we talking?

**Comment by Major-** [comment]
ping @thispixel

**Comment by thispixel** [comment]
slow enough that it will exceed the time set in NetworkConstants
`
	/**
	 * The number of seconds before a connection becomes idle.
	 */
	public static final int IDLE_TIME = 15;
`

**Comment by Major-** [comment]
So is this actually something we should care about, in practice?

**Comment by thispixel** [comment]
Yes specially if people do use the http server for serving the game applet, or anything else.

**Comment by Major-** [comment]
Okay, are you available to/interested in tackling this?

**Comment by thispixel** [comment]
I don't know if just removing the IdleStateHandler from the chain once the request has been read will cause other issues. May have to look into how other types of file serving servers handle it.

**Comment by Major-** [comment]
Basically we want to bump the IDLE_TIME up to something longer (45 or 60 seconds). Removing the IdleStateHandler means we'd be vulnerable to something similar to the slowloris vulnerability apache had a few years ago (although we wouldn't fare as bad).

**Comment by thispixel** [comment]
How does apache / nginx deal with this sort of thing?

**Comment by Major-** [comment]
Both by timing out the connection, similar to apollo, although I don't know how long they wait before doing so (probably longer than the times mentioned above).

**Comment by ryleykimmel** [comment]
We can increase the read timeout without changing the write timeout
